deepmd/utils/tabulate.py

```diff
@@ -82,20 +82,40 @@ def __init__(self,
         self.sub_graph, self.sub_graph_def = self._load_sub_graph()
         self.sub_sess = tf.Session(graph = self.sub_graph)
 
-        try:
-            self.sel_a = self.graph.get_operation_by_name('ProdEnvMatA').get_attr('sel_a')
-            self.prod_env_mat_op = self.graph.get_operation_by_name ('ProdEnvMatA')
-        except Exception:
-            self.sel_a = self.graph.get_operation_by_name('DescrptSeA').get_attr('sel_a')
-            self.prod_env_mat_op = self.graph.get_operation_by_name ('DescrptSeA')
+        if isinstance(self.descrpt, deepmd.descriptor.DescrptSeR):
+            try:
+                self.sel_a = self.graph.get_operation_by_name('ProdEnvMatR').get_attr('sel')
+                self.prod_env_mat_op = self.graph.get_operation_by_name ('ProdEnvMatR')
+            except KeyError:
+                self.sel_a = self.graph.get_operation_by_name('DescrptSeR').get_attr('sel')
+                self.prod_env_mat_op = self.graph.get_operation_by_name ('DescrptSeR')
+        elif isinstance(self.descrpt, deepmd.descriptor.DescrptSeA):
+            try:
+                self.sel_a = self.graph.get_operation_by_name('ProdEnvMatA').get_attr('sel_a')
+                self.prod_env_mat_op = self.graph.get_operation_by_name ('ProdEnvMatA')
+            except KeyError:
+                self.sel_a = self.graph.get_operation_by_name('DescrptSeA').get_attr('sel_a')
+                self.prod_env_mat_op = self.graph.get_operation_by_name ('DescrptSeA')
+        elif isinstance(self.descrpt, deepmd.descriptor.DescrptSeT):
```

Comment:
I think `DescrptSeA` and `DescrptSeT` are the same here?

Reply:
Yes, they are the same. But considering the possible changes later, I have explicitly distinguished them.